File: apps/web/src/code/python/basics/metadata.py
from e2b import Sandbox, SandboxState
from e2b.sandbox.sandbox_api import SandboxQuery

sandbox = Sandbox(
    template="base",
    metadata={"user_id": "uniqueID"},  # $HighlightLine
)

# Keep the sandbox alive for 60 seconds
sandbox.set_timeout(60)
# You can even close the script

# Later, can be even from another process
# List all running sandboxes
paginator = Sandbox.list(query=SandboxQuery(state=[SandboxState.RUNNING]))
running_sandboxes = paginator.next_items()

# Find the sandbox by metadata
for running_sandbox in running_sandboxes:
    if running_sandbox.metadata.get("user_id", "") == "uniqueID":
        sandbox = Sandbox.connect(running_sandbox.sandbox_id)
        break
else:
    # Sandbox not found
    pass

sandbox.close()

File: apps/web/src/code/js/basics/metadata.js
import { Sandbox } from 'e2b'

const sandbox = await Sandbox.create({
  template: 'base',
  metadata: { userID: 'uniqueID' }   // $HighlightLine
})
// Keep the sandbox alive for 60 seconds
await sandbox.keepAlive(60_000)
// You can even close the script

// Later, can be even from another process
// List all running sandboxes
const paginator = Sandbox.list({ query: {state: ['running']}})
const runningSandboxes = await paginator.nextItems()
// Find the sandbox by metadata
const found = runningSandboxes.find(s => s.metadata?.userID === 'uniqueID')
if (found) {
  // Sandbox found, we can reconnect to it
  const sandbox = await Sandbox.reconnect(found.sandboxID)
} else {
  // Sandbox not found
}

await sandbox.close()
